src/app/utilities/AnnotationUtilities.ts

```diff
@@ -0,0 +1,228 @@
+/* Import Types */
+import { Dict } from 'app/Types';
+
+/* Import Sources */
+import DigitalSpecimen from 'sources/dataModel/digital_specimen.json';
+import DigitalEntity from 'sources/dataModel/digital_entity.json';
+import MaterialEntity from 'sources/dataModel/material_entity.json';
+import Identification from 'sources/dataModel/identifications.json';
+import Assertion from 'sources/dataModel/assertions.json';
+import Occurrence from 'sources/dataModel/occurrences.json';
+import EntityRelationship from 'sources/dataModel/entity_relationships.json';
+import Citation from 'sources/dataModel/citations.json';
+import Identifier from 'sources/dataModel/identifiers.json';
+import ChronometricAge from 'sources/dataModel/chronometric_age.json';
+import Location from 'sources/dataModel/location.json';
+
+
+/* Function for pushing to the annotatable properties list, or recall Construct List */
```

Comment:
Could you explain this one to me, trying to wrap my mind around it but having some difficulty

Reply:
Ah yes, so this file has multiple purposes and was created to relieve the template files of all these additional functions.
Primary function of this file is to create the list of annotatable properties and classes, the dropdown lists you will see when opening the Side Panel.
Field properties are grouped by and belong to a class, so simply looping over all the data would not fit the needs, also properties might change by name or classes by structure, hence it start blank with the base files from the data model.

- Main function for creating the lists is ConstructTargetPropertiesLists, which bases the base model (Digital Specimen or Digital Media) and the lists with according function to push to these lists. Then goes on to call ConstructList to actually construct the list from the base model (yes good function names, I know).

- ConstructList goes on to create the full field properties and classes lists. It loops over the different levels in the data model (classes), then calls on PushToClassesList, which checks if we are interacting with a property or a class, based on the type. If it is a string, integer or boolean, it gets pushed to the propertiesList at the right level (called schema and subSchema in code). If it encounters another object, it will go on to create a new level and repeat the process by calling this ConstructList.

- ExtractFromSchema: a function that is used multiple times and also exported from this file for usage elsewhere. It takes a parameter which should point to one of the existing schemas of the data model. If it recognizes one, it will request all the properties from that schema and return them.

Overall it is a lot of code for two simple looking dropdowns, but it also builds the foundation for handling annotations, defining the JSON routes by extracting properties and classes from the different levels of the data model.